service/src/main/kotlin/fi/espoo/evaka/pedagogicaldocument/PedagogicalDocumentController.kt

```diff
@@ -0,0 +1,259 @@
+// SPDX-FileCopyrightText: 2017-2021 City of Espoo
+//
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+package fi.espoo.evaka.pedagogicaldocument
+
+import fi.espoo.evaka.Audit
+import fi.espoo.evaka.shared.AttachmentId
+import fi.espoo.evaka.shared.ChildId
+import fi.espoo.evaka.shared.PedagogicalDocumentId
+import fi.espoo.evaka.shared.auth.AuthenticatedUser
+import fi.espoo.evaka.shared.db.Database
+import fi.espoo.evaka.shared.db.mapColumn
+import fi.espoo.evaka.shared.domain.HelsinkiDateTime
+import fi.espoo.evaka.shared.security.AccessControl
+import fi.espoo.evaka.shared.security.Action
+import org.jdbi.v3.core.kotlin.mapTo
+import org.jdbi.v3.core.result.RowView
+import org.springframework.http.ResponseEntity
+import org.springframework.web.bind.annotation.DeleteMapping
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.PutMapping
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RestController
+
+@RestController
+@RequestMapping("/pedagogical-document")
+class PedagogicalDocumentController(
+    private val accessControl: AccessControl,
+) {
+    @PostMapping
+    fun createPedagogicalDocument(
+        db: Database.Connection,
+        user: AuthenticatedUser,
+        @RequestBody body: PedagogicalDocumentPostBody
+    ): ResponseEntity<PedagogicalDocument> {
+        Audit.PedagogicalDocumentUpdate.log(body.childId)
+        accessControl.requirePermissionFor(user, Action.Child.CREATE_PEDAGOGICAL_DOCUMENT, body.childId.raw)
+        val doc = createDocument(db, user, body)
+        return ResponseEntity.ok(doc)
+    }
+
+    @PutMapping("/{documentId}")
+    fun updatePedagogicalDocument(
+        db: Database.Connection,
+        user: AuthenticatedUser,
+        @PathVariable documentId: PedagogicalDocumentId,
+        @RequestBody body: PedagogicalDocumentPostBody
+    ): ResponseEntity<PedagogicalDocument> {
+        Audit.PedagogicalDocumentUpdate.log(documentId)
+        accessControl.requirePermissionFor(user, Action.Child.UPDATE_PEDAGOGICAL_DOCUMENT, body.childId.raw)
```

Comment:
Could the access control scope be the document instead of child? In pretty much all other access control checks we use the most specific ID. For example, placement access control uses placement ID, application access control uses application ID, etc...so pedagogical document access control probably could use pedagogical document ID.

In the create endpoint we of course need to use child ID since we don't have a document, but in all other cases a document ID is available and it's more complicated and error-prone to manually fetch the child id for access control checks in multiple places.